app/server/lib/FlexServer.ts

```diff
@@ -442,24 +442,33 @@ export class FlexServer implements GristServer {
 
   public addLogging() {
     if (this._check('logging')) { return; }
-    if (process.env.GRIST_LOG_SKIP_HTTP) { return; }
+    if (isAffirmative(process.env.GRIST_LOG_SKIP_HTTP)) { return; }
```

Comment:
This looks like normally http logs are emitted? Is that consistent with the documentation, or am I confusing myself with the negations?

Reply:
### Before this PR

The trick here is that currently `GRIST_LOG_SKIP_HTTP` should be set to:
 - `""` to enable logging the http requests;
 - anything else (even to `false`) to disable the http requests;

By default, the variable is set to `true` ([source](https://github.com/gristlabs/grist-core/blob/main/stubs/app/server/server.ts#L18)), so the requests are not logged.

### With this PR

As the expected values are confusing, I change the behavior so the variable is expected to be a boolean-ish, and continue to equal to `true` if unset.

### Proposal

I agree that the naming is confusing. As suggested [in my response to your remark in the other thread](https://github.com/gristlabs/grist-core/pull/913#discussion_r1726561553), I write here a detailed proposal:

 - `GRIST_LOG_SKIP_HTTP` continue to exist, but remains undocumented and is marked as deprecated, suggesting the admins to stop using this variable in favor of the other one detailed below.
   - also I remove this change, the expected values for `GRIST_LOG_SKIP_HTTP` are the one before this PR;
 - introduce the `GRIST_LOG_HTTP`, document it, make it expect boolean-ish values (so use `isAffirmative(process.env.GRIST_LOG_HTTP)` to check whether we should log the requests.
   - by default, this variable is unset, so we don't log anything

Interactions between `GRIST_LOG_SKIP_HTTP` and `GRIST_LOG_HTTP` may be tricky, so here is a table with the possible values:

| GRIST_LOG_SKIP_HTTP | GRIST_LOG_HTTP | Request logged? | Remarks |
|:---:|:---:|:---:|:---:|
|unset|unset| NO | -- |
|"true"|unset| NO | -- |
|"false" / "0" / "no" / ... |unset| NO | as per the behavior currently in production |
|"" |unset| YES |  As per the behavior currently in production. <br>A deprecation warning is logged |
|unset|"true" / "1" / "yes" / "on" | **YES** | -- |
|set to any value ("true", "whatever", "false") |"true" / "1" / "yes" / "on" | **YES** | `GRIST_LOG_HTTP` has priority over `GRIST_LOG_SKIP_HTTP`, assume that the latest variable is activated because it is [its default value](https://github.com/gristlabs/grist-core/blob/main/stubs/app/server/server.ts#L18) |
|""|"true" / "1" / "yes" / "on" | **YES** | A deprecation warning is logged |
|""|"false" | -- | The server crashes explaining the inconsistency |

Reply:
I like this better. Here's a condensed version of the same table:

| `GRIST_LOG_SKIP_HTTP` | `GRIST_LOG_HTTP` | Request logged? | Remarks |
|:---:|:---:|:---:|:---:|
|unset or non-empty string value|	unset or "false" or any non-truthy value|	NO	
|""|	unset|	YES|	As per the behavior currently in production.
|""|	"false" or any non-truthy value|	--	|The server crashes explaining the inconsistency
|unset or non-empty string value|	"true" / "1" / "yes" / "on"	|YES	|GRIST_LOG_HTTP has priority over GRIST_LOG_SKIP_HTTP, since the latter may be merely set to its [its default value](https://github.com/gristlabs/grist-core/blob/main/stubs/app/server/server.ts#L18)
|""|	"true" / "1" / "yes" / "on"	|YES	

(BTW, I looked at this table in Grist -- thank you for the "Use as table headers" feature! And then I was sad I couldn't "Export as markdown table" 🙂 )

Reply:
Done in a6dbe91d6e02f4a14dca31d624537736d71c0f26.

I managed to improve a bit the behavior: I removed the default value for `GRIST_LOG_SKIP_HTTP` (the default behavior now thanks to `GRIST_LOG_HTTP` is to disable the requests logging). So I make the server crash whenever both of these two variables are set.

The actualized version of the above table is the following:

| `GRIST_LOG_SKIP_HTTP`           | `GRIST_LOG_HTTP`                         | Request logged? | Remarks                                                                                                                                                                                               |
| ------------------------------- | ---------------------------------------- | --------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| unset or non-empty string value | unset or "false" or any non-truthy value | NO              |  `GRIST_LOG_SKIP_HTTP` [does not equal to `true` by default anymore](https://github.com/gristlabs/grist-core/pull/913/files#diff-180fe6da63030c6e6cc8b6c579dd8e444500a60a2a1f353cf23d7547b43dca20L18), but leaving `GRIST_LOG_HTTP` unset prevents to log the requests                                                                                                                                                     |
| ""                              | unset                                    | YES             | As per the behavior currently in production.                                                                                                                                                          |
| "" or non-empty string value                             | "false" or any non-truthy value          | \--             | The server crashes asking to set only the `GRIST_LOG_HTTP` variable |
| unset | "true" / "1" / "yes" / "on"              | YES             | |

Also any time the `GRIST_LOG_SKIP_HTTP` option is set, a deprecation warning is logged, suggesting to use `GRIST_LOG_HTTP` instead.